test/demo-extension/background-handlers.ts

```diff
@@ -0,0 +1,52 @@
+import {isBackgroundPage} from 'webext-detect-page';
+
+import {Contract} from '../..';
+
+export const getExtensionIdContract: Partial<
+  Contract<string, typeof getExtensionId>
+> = {
+  type: 'getExtensionId',
+};
+
+export const sumContract: Partial<Contract<string, typeof sum>> = {
+  type: 'sum',
+};
+
+export const notRegisteredContract: Partial<
+  Contract<string, () => Promise<never>>
+> = {
+  type: 'notRegistered',
+};
+
+export const backgroundOnlyContract: Partial<
+  Contract<string, typeof backgroundOnly>
+> = {
+  type: 'backgroundOnly',
+};
+
+export const throwsContract: Partial<Contract<string, typeof throws>> = {
+  type: 'throws',
+};
+
+export async function getExtensionId(): Promise<string> {
+  return chrome.runtime.id;
+}
+
+export async function sum(
+  _: browser.runtime.MessageSender,
+  ...addends: number[]
+): Promise<number> {
+  return addends.reduce((a, b) => a + b);
+}
```

Comment:
_(I'll comment in review threads so that the conversation is easier to follow)_

> If you write as a constant you get a bit better error message
> 
> ![image](https://user-images.githubusercontent.com/1879821/131594553-dfd2b753-b425-4661-a8c1-af54b5839772.png)
> 
> It's saying that the type is wrong because if: 1) I have something of type message, 2) I can't substitute sumMethod for it because the args might not be numbers



Indeed, as shown by your example:

```ts
async function inner(
  _: browser.runtime.MessageSender,
  ...addends: number[]
): Promise<number> {
	return addends.reduce((a, b) => a + b);
};
const outer: Method = inner;
```

This essentially tells tsc to discard any types of `inner` and just set `outer` to `Method`, so `outer` has no `number` parameters when called.

In practice this isn't a problem because `outer` is called with whatever `onMessage` receives.

What's important is that:

- `inner` (the "method") has the right signature for `meta`, which it does (this is enforced by `addHandler`’s types)
- `outer` (the "public method") has the right interface, because that's what the user calls, and that works even if I specify `any[]`, as you suggested 🎉 

	<img width="445" alt="Screen Shot 2" src="https://user-images.githubusercontent.com/1402241/131650371-1754d80a-0c09-40e2-95e3-4f6e9be747c3.png">